### ISSUE TITLE
Add function `ispackage` to stdlib `inspect`

### ISSUE THREAD

**Issue by Xiaokang2022** [python/cpython#125633]
# Feature or enhancement

### Proposal:

Currently, the built-in module `inspect` lacks a function to determine whether an object is a package or not. This is a small but useful feature.

According to the documentation, this feature should not be difficult to implement, and we should add it to the standard library:

1. [Definition of a package](https://docs.python.org/3/reference/import.html#packages)
2. [module.\_\_package\_\_](https://docs.python.org/3/reference/datamodel.html#module.__package__)
3. [module.\_\_name\_\_](https://docs.python.org/3/reference/datamodel.html#module.__name__)
4. [module.\_\_spec\_\_](https://docs.python.org/3/reference/datamodel.html#module.__spec__)

I've tried to implement this feature and will make a PR later.

### Has this already been discussed elsewhere?

This is a minor feature, which does not need previous discussion elsewhere

### Linked PRs
* gh-125634

**Comment by Xiaokang2022** [comment]
My idea for implementing this feature is:

```mermaid
flowchart LR
    1(object)
    2{is module?}
    3{is __ main __?}
    4{__ spec __.parent 
    == __ spec __.name?}
    5(package)

    1 --> 2 --Y--> 3 --N--> 4 --Y--> 5
```

**Comment by ncoghlan** [comment]
PR has been merged for inclusion in Python 3.14.0a2.
